store/state/enum.go

```diff
@@ -0,0 +1,37 @@
+package state
+
+type Enum struct {
+	Value
+}
+
+func NewEnum(v Value) Enum {
+	return Enum{v}
+}
+
+func (v Enum) Get(ctx Context) (res byte) {
+	v.Value.Get(ctx, &res)
+	return
+}
+
+func (v Enum) GetSafe(ctx Context) (res byte, err error) {
+	err = v.Value.GetSafe(ctx, &res)
+	return
+}
+
+func (v Enum) Set(ctx Context, value byte) {
+	v.Value.Set(ctx, value)
+}
+
+func (v Enum) Incr(ctx Context) (res byte) {
+	res = v.Get(ctx) + 1
+	v.Set(ctx, res)
+	return
+}
+
+func (v Enum) Transit(ctx Context, from, to byte) bool {
```

Comment:
What is this used for?

Reply:
`Transit` is used for moving the stored state from `from` to `to`. Mostly used for the handshaking process in connection and channel. Named `Enum` because it is essentially a `byte` and considered as an enum type. Can be changed.